src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs

```diff
@@ -249,5 +258,42 @@ private static ParameterInfo GetParameterInfo(INamedTypeSymbol type, bool isArra
         {
             return ParameterInfo.GetParameterInfo(type, isArray, arrayRank, isParams);
         }
+
+        private static bool IsValidToStringCall(IInvocationOperation invocationOperation, ImmutableHashSet<INamedTypeSymbol> invariantToStringTypes,
+            INamedTypeSymbol? dateTimeType, INamedTypeSymbol? dateTimeOffsetType, INamedTypeSymbol? timeSpanType)
+        {
+            var targetMethod = invocationOperation.TargetMethod;
+
+            if (targetMethod.Name != "ToString")
+            {
+                return false;
+            }
+
+            if (invariantToStringTypes.Contains(targetMethod.ContainingType))
+            {
+                return true;
+            }
+
+            if (invocationOperation.Arguments.Length != 1 ||
+                !invocationOperation.Arguments[0].Value.ConstantValue.HasValue ||
+                !(invocationOperation.Arguments[0].Value.ConstantValue.Value is string format))
```

Comment:
`Arguments[0]` does not mean it is the argument for the first parameter. If named arguments are used, such as you specify the argument to second non-first parameter first, then it will be that argument. You want to find argument that corresponds the first parameter here, and would need to iterate through all arguments till to you find one with matching parameter. Can you define an extension method `IArgumentOperation FindArgumentForParameterAtIndex(this ImmutableArray<IArgumentOperation> arguments, int parameterIndex)` and use it here?

Reply:
Ah actually, `invocationOperation.Arguments.Length != 1` check ensures you are good. Ignore my suggestion.